ci(tools): fix iroha AIO image build flake on CI

**Describe the bug**

`DOCKER_BUILDKIT=1 docker build ./tools/docker/iroha-all-in-one/ -f ./tools/docker/iroha-all-in-one/Dockerfile`

The image build fails randomly within the CI.

**To Reproduce**

Run the CI repeatedly and at some point it will fail to build the containers due to the Iroha AIO image not building.

**Expected behavior**

Build is stable.

**Logs/Stack traces**

Full logs: [2021-11-23-cactus-ci-log-build-containers-iroha-crash.txt](https://github.com/hyperledger/cactus/files/7591378/2021-11-23-cactus-ci-log-build-containers-iroha-crash.txt)

Relevant section of logs:

```sh
#8 2476.9 Computing installation plan...
#8 2476.9 The following packages will be built and installed:
#8 2476.9     rxcpp[core]:x64-linux (from HEAD)
#8 2476.9 Detecting compiler hash for triplet x64-linux...
#8 2477.5 Starting package 1/1: rxcpp:x64-linux
#8 2477.5 Building package rxcpp[core]:x64-linux from HEAD...
#8 2477.6 -- Downloading https://api.github.com/repos/ReactiveX/RxCpp/git/refs/heads/master...
#8 2477.6 -- Downloading https://api.github.com/repos/ReactiveX/RxCpp/git/refs/heads/master... Failed. Status: 22;"HTTP response code said error"
#8 2477.6 CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:181 (message):
#8 2477.6       
#8 2477.6       Failed to download file.
#8 2477.6       If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
#8 2477.6       variables to "***your-proxy-ip-address:port/".
#8 2477.6       Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues
#8 2477.6 
#8 2477.6 Call Stack (most recent call first):
#8 2477.6   scripts/cmake/vcpkg_from_github.cmake:174 (vcpkg_download_distfile)
#8 2477.6   ports/rxcpp/portfile.cmake:1 (vcpkg_from_github)
#8 2477.6   scripts/ports.cmake:79 (include)
#8 2477.6 
#8 2477.6 
#8 2477.6 Error: Building package rxcpp:x64-linux failed with: BUILD_FAILED
#8 2477.6 Please ensure you're using the latest portfiles with `./vcpkg update`, then
#8 2477.6 submit an issue at https://github.com/Microsoft/vcpkg/issues including:
#8 2477.6   Package: rxcpp:x64-linux
#8 2477.6   Vcpkg version: 2020.06.15-unknownhash
#8 2477.6 
#8 2477.6 Additionally, attach any relevant sections from the log files above.
#8 ERROR: executor failed running [/bin/sh -c iroha/vcpkg/build_iroha_deps.sh && vcpkg/vcpkg integrate install]: exit code: 123
------
 > [builder 4/7] RUN iroha/vcpkg/build_iroha_deps.sh && vcpkg/vcpkg integrate install:
------
executor failed running [/bin/sh -c iroha/vcpkg/build_iroha_deps.sh && vcpkg/vcpkg integrate install]: exit code: 123
```

**Cloud provider or hardware configuration:**

Github Actions Runner - Managed 

**Operating system name, version, build:**

Ubuntu 20.04

**Hyperledger Cactus release version or commit (git rev-parse --short HEAD):**

`1.0.0-rc.2`

**Hyperledger Cactus Plugins/Connectors Used**

N/A

**Additional context**

This came up randomly during the CI execution of an unrelated pull request.

--- a/tools/docker/iroha-all-in-one/Dockerfile
+++ b/tools/docker/iroha-all-in-one/Dockerfile
@@ -2,11 +2,18 @@ FROM ubuntu:20.04 as builder
 ARG DEBIAN_FRONTEND=noninteractive
 
 RUN set -e && apt-get update && apt-get install -y --no-install-recommends \
-    file build-essential ninja-build git ca-certificates tar curl unzip cmake pkg-config zip
-RUN git clone https://github.com/hyperledger/iroha.git -b 1.2.0 
-RUN iroha/vcpkg/build_iroha_deps.sh && vcpkg/vcpkg integrate install
+    file build-essential ninja-build ca-certificates tar curl unzip cmake pkg-config zip software-properties-common
+
+RUN add-apt-repository ppa:git-core/ppa
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends git
+
+RUN git clone https://github.com/hyperledger/iroha.git -b 1.4.0 
+RUN iroha/vcpkg/build_iroha_deps.sh
+RUN /vcpkg-build/vcpkg integrate install
 WORKDIR /iroha/build/
-RUN cmake -DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TYPE=Release -DPACKAGE_DEB=ON -G "Ninja" ..
+
+RUN cmake -DCMAKE_TOOLCHAIN_FILE=/vcpkg-build/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TYPE=Release -DPACKAGE_DEB=ON -G "Ninja" ..
 RUN cmake --build . --target package -- -j$(nproc)
 
 FROM ubuntu:20.04

